### ISSUE TITLE
[BUG] It's not possible to have workspaces with different versions of React

### ISSUE THREAD

**Issue by jfulse** [npm/cli#6040]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

AFAIK It's not possible to have different workspaces with different versions of React due to hoisting.

### Expected Behavior

It would be nice to be to have different versions of React in different workspaces, and upgrade different parts of the app at different times

### Steps To Reproduce

- create a new npm package with two workspaces `packages/a` and `packages/b`; its `package.json` including `workspaces: ["packages/*"]`
- create a new app in each of the workspaces using `create-react-app`
- Degrade the version in workspace `a` down to 17 by editing its `package.json`
- Add a hook in each app, which will error if there are two versions running
- Remove all `node_modules` folders, delete all `package-lock.json` files, and run `npm i` in the root folder
- Start the two apps. This gives the "several versions of react" error with the hooks. Examining the `node_modules` folder in the root level we see that it contains one of the react versions.
- We also tried running `npm i` in each of the workspaces separately with the same result

In addition we tried creating a similar setup but without the full app, only installing (different versions of) `react` and `react-dom` in the two; still the root level `node_modules` contains one of them.

### Environment

- npm: 8.12.1
- Node.js: v16.13.1
- OS Name: macOS Bug Sur
- System Model Name: Macbook Pro
- npm config:
```ini
; "user" config from /Users/[REDACTED]/.npmrc

//registry.npmjs.org/:_authToken = (protected)
access = "public"

; "project" config from /Users/[REDACTED]repos/[REDACTED]/.npmrc

engine-strict = true

; node bin location = /Users/[REDACTED]/.nvm/versions/node/v16.13.1/bin/node
; node version = v16.13.1
; npm local prefix = /Users/[REDACTED]/repos/libry-content
; npm version = 8.12.1
; cwd = /Users/[REDACTED]/repos/[REDACTED]
; HOME = /Users/[REDACTED]
; Run `npm config ls -l` to show all defaults.
```

**Comment by fritzy** [comment]
We're working on an `--install-strategy=linked` that will address this. It might also be worth directly addressing workspace without having a new install strategy. In the meantime, this is actively being addressed.

**Comment by jamiebarrow** [comment]
Nice to see that this is now available in npm 9.4.0 :)

https://nodejs.org/en/blog/release/v19.6.0/#upgrade-npm-to-9-4-0

**Comment by eng-almeida** [comment]
@fritzy Thank you for the heads up 🙌 I'm facing the exact same issue described in the issue. I installed latest npm, 9.8.1, but when I run `npm i --install-strategy=linked` I'm getting the following error:

```
warn reify The "linked" install strategy is EXPERIMENTAL and may contain bugs.
timing command:i Completed in 2774ms
verbose stack TypeError: Cannot read properties of undefined (reading 'path')
verbose stack     at processDeps (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/isolated-reifier.js:405:28)
verbose stack     at processEdges (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/isolated-reifier.js:434:9)
verbose stack     at processEdges (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/isolated-reifier.js:427:9)
verbose stack     at processEdges (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/isolated-reifier.js:427:9)
verbose stack     at processEdges (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/isolated-reifier.js:427:9)
verbose stack     at [createIsolatedTree] (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/isolated-reifier.js:443:5)
verbose stack     at async Arborist.reify (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:167:24)
verbose stack     at async Install.exec (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/lib/commands/install.js:149:5)
verbose stack     at async module.exports (/Users/[REDACTED]/.nvm/versions/node/v18.15.0/lib/node_modules/npm/lib/cli-entry.js:61:5)
```
Any hints? 🙏  Thank you!
